napari/utils/events/_tests/test_evented_list.py

```diff
@@ -0,0 +1,342 @@
+
+@pytest.mark.parametrize(
+    'sources,dest,expectation',
+    [
+        ((2,), 0, [2, 0, 1, 3, 4, 5, 6, 7]),  # move single item
+        ((2, 4), -2, [0, 1, 3, 5, 6, 2, 4, 7]),  # negative indexing
+        ([0, 2, 3], 6, [1, 4, 5, 0, 2, 3, 6, 7]),  # move back
+        ([4, 7], 1, [0, 4, 7, 1, 2, 3, 5, 6]),  # move forward
+        ([0, 5, 6], 3, [1, 2, 0, 5, 6, 3, 4, 7]),  # move in between
+        ([slice(None, 3)], 6, [3, 4, 5, 0, 1, 2, 6, 7]),  # move slice back
+        ([slice(5, 8)], 2, [0, 1, 5, 6, 7, 2, 3, 4]),  # move slice forward
+        ([slice(1, 8, 2)], 3, [0, 2, 1, 3, 5, 7, 4, 6]),  # move slice between
+        ([slice(None, 8, 3)], 4, [1, 2, 0, 3, 6, 4, 5, 7]),
+        ([0, 2, 3, 2, 3], 6, [1, 4, 5, 0, 2, 3, 6, 7]),  # strip dupe indices
+        ([slice(None, 8, 3), 0, 3, 6], 4, [1, 2, 0, 3, 6, 4, 5, 7]),
+    ],
+)
+def test_move_multiple(sources, dest, expectation):
+    """Test the that we can move objects with the move method"""
+    el = EventedList(range(8))
+    el.events = Mock(wraps=el.events)
+    assert el == [0, 1, 2, 3, 4, 5, 6, 7]
+
+    def _fail(e):
+        raise AssertionError("unexpected event called")
+
+    el.events.removing.connect(_fail)
+    el.events.removed.connect(_fail)
+    el.events.inserting.connect(_fail)
+    el.events.inserted.connect(_fail)
+
+    el.move_multiple(sources, dest)
+    assert el == expectation
+    el.events.moving.assert_called_once()
+    el.events.moved.assert_called_once()
+    el.events.reordered.assert_called_with(value=expectation)
+
+
+def test_move_multiple_mimics_slice_reorder():
+    """Test the that move_multiple provides the same result as slice insertion.
+    """
+    data = list(range(8))
+    el = EventedList(data)
+    el.events = Mock(wraps=el.events)
+    assert el == data
+    new_order = [1, 5, 3, 4, 6, 7, 2, 0]
+    # this syntax
+    el.move_multiple(new_order, 0)
+    # is the same as this syntax
+    data[:] = [data[i] for i in new_order]
+    assert el == new_order
+    assert el == data
+    el.events.moving.assert_called_with(index=new_order, new_index=0)
+    el.events.moved.assert_called_with(
+        index=new_order, new_index=0, value=new_order,
+    )
+    el.events.reordered.assert_called_with(value=new_order)
+
+    # move_multiple also works omitting the insertion index
+    el[:] = list(range(8))
+    el.move_multiple(new_order) == [el[i] for i in new_order]
+
+
+def test_slice(test_list, regular_list):
+    """Slicing an evented list should return a same-class evented list."""
+    test_slice = test_list[1:3]
+    regular_slice = regular_list[1:3]
+    assert tuple(test_slice) == tuple(regular_slice)
+    assert isinstance(test_slice, test_list.__class__)
+
+
+NEST = [0, [10, [110, [1110, 1111, 1112], 112], 12], 2]
+
+
+def test_nested_indexing():
+    """test that we can index a nested list with nl[1, 2, 3] syntax."""
+    ne_list = NestableEventedList(NEST)
+    indices = [tuple(int(x) for x in str(n)) for n in flatten(NEST)]
```

Comment:
```suggestion
    # 110 -> '110' -> (1, 1, 0)
    indices = [tuple(int(x) for x in str(n)) for n in flatten(NEST)]
```

it took me a bit to grok this expression!